src/unix/thread.c

```diff
@@ -203,6 +203,8 @@ void thread_sleep_interruptible(void)
     disable_interrupts();
     assert(current->blocked_on);
     thread_log(current, "sleep interruptible (on \"%s\")", blockq_name(current->blocked_on));
+    ftrace_thread_switch(current, 0);
+    set_current_thread(0);
```

Comment:
`set_current_thread(0)` shouldn't be necessary, since it's already being done when entering the runloop (by calling sched_thread_pause() which in turn calls thread_pause())

Reply:
Thanks for the catch!